Engine/Module/ModuleSpacePartitioning.cpp

```diff
@@ -0,0 +1,267 @@
+
+		float maxX = std::fmax(mesh->owner->aabb.bounding_box2D.maxPoint.x, global_AABB.maxPoint.x);
+		float maxY = std::fmax(mesh->owner->aabb.bounding_box2D.maxPoint.y, global_AABB.maxPoint.y);
+		global_AABB.maxPoint = float2(maxX, maxY);
+		global_AABB.minPoint = float2(minX, minY);
+	}
+
+	ol_quadtree->Create(global_AABB);
+	for (const auto& mesh : App->renderer->meshes)
+	{
+		ol_quadtree->Insert(*mesh->owner);
+	}
+}
+
+void ModuleSpacePartitioning::GenerateOctTree()
+{
+
+	BROFILER_CATEGORY("Generate OctTree", Profiler::Color::Lavender);
+	AABB global_AABB;
+	global_AABB.SetNegativeInfinity();
+
+	for (const auto& mesh : App->renderer->meshes)
+	{
+		float minX = std::fmin(mesh->owner->aabb.bounding_box.minPoint.x, global_AABB.minPoint.x);
+		float minY = std::fmin(mesh->owner->aabb.bounding_box.minPoint.y, global_AABB.minPoint.y);
+		float minZ = std::fmin(mesh->owner->aabb.bounding_box.minPoint.z, global_AABB.minPoint.z);
+
+		float maxX = std::fmax(mesh->owner->aabb.bounding_box.maxPoint.x, global_AABB.maxPoint.x);
+		float maxY = std::fmax(mesh->owner->aabb.bounding_box.maxPoint.y, global_AABB.maxPoint.y);
+		float maxZ = std::fmax(mesh->owner->aabb.bounding_box.maxPoint.z, global_AABB.maxPoint.z);
+
+		global_AABB.maxPoint = float3(maxX, maxY, maxZ);
+		global_AABB.minPoint = float3(minX, minY, minZ);
+	}
+
+	ol_octtree->Create(global_AABB);
+	for (const auto&  mesh : App->renderer->meshes)
+	{
+		ol_octtree->Insert(*mesh->owner);
+	}
+}
+
+void ModuleSpacePartitioning::InsertAABBTree(GameObject* game_object)
+{
+	ComponentMeshRenderer* object_mesh = static_cast<ComponentMeshRenderer*>(game_object->GetComponent(Component::ComponentType::MESH_RENDERER));
+	if (object_mesh != nullptr)
+	{
+		ol_abbtree->Insert(game_object);
+	}
+}
+
+void ModuleSpacePartitioning::RemoveAABBTree(GameObject* game_object)
+{
+	ComponentMeshRenderer* object_mesh = static_cast<ComponentMeshRenderer*>(game_object->GetComponent(Component::ComponentType::MESH_RENDERER));
+	if (object_mesh != nullptr)
+	{
+		ol_abbtree->Remove(game_object);
+	}
+}
+
+void ModuleSpacePartitioning::UpdateAABBTree(GameObject * game_object)
+{
+	if(game_object->IsStatic())
+	{
+		return;
+	}
+
+	ComponentMeshRenderer* object_mesh = static_cast<ComponentMeshRenderer*>(game_object->GetComponent(Component::ComponentType::MESH_RENDERER));
+	if (object_mesh != nullptr)
+	{
+		ol_abbtree->UpdateObject(game_object);
+	}
+}
+
+void ModuleSpacePartitioning::ResetAABBTree()
+{
+	ol_abbtree.reset(new OLAABBTree(INITIAL_SIZE_AABBTREE));
+}
+
+void ModuleSpacePartitioning::DrawAABBTree() const
+{
+	BROFILER_CATEGORY("Render AABBTree", Profiler::Color::Lavender);
+
+	ol_abbtree->Draw();
+}
+
+void ModuleSpacePartitioning::GetCullingMeshes(const ComponentCamera * camera) const
+{
+	BROFILER_CATEGORY("Get culling meshes", Profiler::Color::Lavender);
+
+	App->renderer->meshes_to_render.clear();
+	switch (App->debug->culling_mode)
```

Comment:
Same as previous comment, culling mode should be inside this module as is no more a debug feature